src/IECoreScene/ShaderNetworkAlgo.cpp

```diff
@@ -684,6 +713,213 @@ IECore::DataPtr loadSpline(
+
+		for( const auto &i : origParameters )
+		{
+			if( const SplinefColor3fData *colorSpline = runTimeCast<const SplinefColor3fData>( i.second.get() ) )
+			{
+				expandSpline( i.first, colorSpline->readable(), newParameters );
+			}
+			else if( const SplineffData *floatSpline = runTimeCast<const SplineffData>( i.second.get() ) )
+			{
+				expandSpline( i.first, floatSpline->readable(), newParameters );
+			}
+			else
+			{
+				newParameters[ i.first ] = i.second;
+			}
+		}
+
+		currentSplineArrayAdapters.clear();
+
+
+		std::vector< ShaderNetwork::Connection > connectionsToAdd;
+		ShaderNetwork::ConnectionRange inputConnections = network->inputConnections( s.first );
+		for( ShaderNetwork::ConnectionIterator it = inputConnections.begin(); it != inputConnections.end(); )
+		{
+			// Copy and increment now so we still have a valid iterator
+			// if we remove the connection.
+			const ShaderNetwork::Connection connection = *it++;
+
+			const std::string &destName = connection.destination.name.string();
+			boost::cmatch splineElementMatch;
+			if( !boost::regex_match( destName.c_str(), splineElementMatch, g_splineElementRegex) )
+			{
+				continue;
+			}
+
+			IECore::InternedString parameterName( &destName[splineElementMatch.position(1)], splineElementMatch.length(1) );
+			auto findParameter = origParameters.find( parameterName );
+			if( findParameter == origParameters.end() )
+			{
+				continue;
+			}
+
+			const SplinefColor3fData* colorSplineData = runTimeCast<const SplinefColor3fData>( findParameter->second.get() );
+			const SplineffData* floatSplineData = runTimeCast<const SplineffData>( findParameter->second.get() );
+
+			if( !( colorSplineData || floatSplineData ) )
+			{
+				continue;
+			}
+
+			InternedString adapterHandle;
+			int knotOffset;
+
+			// Insert a conversion shader to handle connection to component
+			auto inserted = currentSplineArrayAdapters.insert( { parameterName, std::make_pair( IECore::InternedString(), 0 ) } );
+			if( inserted.second )
+			{
+				IECore::InternedString splineValuesName = parameterName.string() + "Values";
+				auto findValues = newParameters.find( splineValuesName );
+				ShaderPtr adapter;
+				int adapterArraySize = 0;
+				if( colorSplineData )
+				{
+					knotOffset = getEndPointDuplication( colorSplineData->readable().basis ).first;
+					adapter = new Shader( "Utility/ColorArrayInputAdapter", "osl:shader" );
+
+					const Color3fVectorData *splineValuesData = findValues != newParameters.end() ? runTimeCast<const Color3fVectorData>( findValues->second.get() ) : nullptr;
+					if( !splineValuesData )
+					{
+						throw IECore::Exception( "Internal failure in convertToOSLConventions - expandSpline did not create values." );
+					}
+
+					const std::vector< Imath::Color3f > &splineValues = splineValuesData->readable();
+
+					adapterArraySize = splineValues.size();
+
+					if( adapterArraySize <= maxArrayInputAdapterSize )
```

Comment:
What if it's not? Feels like we'd want at least a warning?

Reply:
Oh, I see we're dealing with that lower down. Feels a bit confusing to separate the two - another thing that could be simplified by the templating suggestion.